lib/elixir/lib/range.ex

```diff
@@ -40,7 +40,8 @@ defmodule Range do
   defstruct first: nil, last: nil
 
   @type t :: %Range{}
-  @type t(first, last) :: %Range{first: first, last: last}
+  @type t(first :: integer, last :: integer) :: %Range{first: integer, last: integer}
```

Comment:
It should either be just 
```elixir
@type t :: %Range{first: integer, last: integer}
```

or

```elixir
@type t :: %Range{first: integer, last: integer}
@type t(first, last) :: %Range{first: first, last: last}
```

I don't think 

```elixir
@type t(first :: integer, last :: integer) :: %Range{first: integer, last: integer}
```

would make sense because you add parameters to the type but then don't use the parameters.